File: obg.Domain/Entities/PurchaseLine.cs
using System.ComponentModel.DataAnnotations;

namespace obg.Domain.Entities
{
    public class PurchaseLine
    {
        [Key] public string IdPurchaseLine { get; set; }
        public string MedicineCode { get; set; }
        public int MedicineQuantity { get; set; } 

        public PurchaseLine() { }
        public PurchaseLine(string idPurchaseLine, string medicineCode, int medicineQuantity)
        {
            IdPurchaseLine = idPurchaseLine;
            MedicineCode = medicineCode;
            MedicineQuantity = medicineQuantity;
        }
    }
}

File: ObgApp/src/app/models/purchaseLine.ts
export enum PurchaseLineStatus {
    accepted,
    rejected,
    unResolved
}

export class PurchaseLine {

    idPurchaseLine: string | null;
    medicineCode: string | null;
    medicineQuantity: number | null;
    tatus: PurchaseLineStatus | null;

    constructor(idPurchaseLine: string | null, medicineCode: string | null, medicineQuantity: number | null, status: PurchaseLineStatus | null) {
        this.idPurchaseLine = idPurchaseLine;
        this.medicineCode = medicineCode;
        this.medicineQuantity = medicineQuantity;
        this.tatus = status;
    }
}
